symbolic-debuginfo/src/sourcebundle.rs

```diff
@@ -938,11 +938,19 @@ where
                 continue;
             }
 
-            let source = if (filename.starts_with('<') && filename.ends_with('>')) || !filter(&file)
-            {
+            let source = if filename.starts_with('<') && filename.ends_with('>') {
                 None
             } else {
-                std::fs::read(&filename).ok()
+                let source_from_object = session
+                    .source_by_path(&filename)
+                    .map_err(|e| SourceBundleError::new(SourceBundleErrorKind::BadDebugFile, e))?;
+                if !filter(&file, &source_from_object) {
+                    None
+                } else {
+                    // Note: we could also use source code directly from the object, but that's not
+                    // what happened here previously - only collected locally present files.
+                    std::fs::read(&filename).ok()
+                }
```

Comment:
I think these branches are better switched.
```suggestion
                if filter(&file, &source_from_object) {
                    // Note: we could also use source code directly from the object, but that's not
                    // what happened here previously - only collected locally present files.
                    std::fs::read(&filename).ok()
                } else {
                    None
                }
```